### PR TITLE
Add base support for Heze block production

### PR DESCRIPTION
**Pull request by conache**
## Proposed Changes

- Enable Heze block production by constructing `BeaconBlockHeze` in `complete_partial_beacon_block_gloas` (the body is identical to the Gloas beacon block variant)
- This work is a pre-requisite for testing the upcoming FOCIL PRs
- Route Heze block building through the existing Gloas `getPayload` path, because the [engine API spec for Bogota](https://github.com/ethereum/execution-apis/blob/main/src/engine/bogota.md) defines no new `getPayload` version for this fork. Heze responses are deliberately deserialized into the Gloas containers (see [TODO added](https://github.com/sigp/lighthouse/pull/9714/changes#diff-8b79597c7e1a33167a01f676e50f09dbc40e761dad5b294086ebda40828e1641R1064-R1065)), so the Heze-specific engine types stay unused until the payloads diverge. Likewise, envelope submission still uses the Gloas `newPayloadV5` path
- Add test producing and importing blocks across the Heze boundary epoch

This PR unblocks Heze testing through the beacon chain harness. Previously, with block production stubbed, the harness could not build or advance a Heze chain. This is a blocker for any Heze integration tests, a prerequisite for the upcoming FOCIL PRs' tests.

### CODE REVIEW

**Review by rahulbarmann — COMMENTED** [review]
LGTM!      
One small coordination note: this conflicts with #9729. That PR splits the bid field by fork, so `BeaconBlockBodyHeze` ends up requiring `SignedExecutionPayloadBidHeze<E>`, while its self-build path still constructs an `ExecutionPayloadBidGloas`. It also leaves `"Block production disabled for Heze"` in place, which is the same part this PR replaces.      
      
Neither branch contains the other, so whichever merges second will need to wire Heze bid construction and populate `inclusion_list_bits`.

**Comment by dapplion** [comment]
🤖 automated (dapplion's agent): First pass review. Nothing blocking, but a few things worth a note.

**`getPayload` for Heze is typed as Gloas.** The `ForkName::Gloas | ForkName::Heze` arm in `get_payload_v6` deserializes the Heze response into `JsonGetPayloadResponseGloas`, so it comes back as `GetPayloadResponse::Gloas` and flows on through `BlockProposalContentsGloas` / `ExecutionPayloadGloas`. That leaves the already-existing `JsonGetPayloadResponseHeze`, `GetPayloadResponseHeze` and `ExecutionPayloadHeze` unreachable on the client side: the mock EL serializes `JsonGetPayloadResponseHeze` and we parse it back as Gloas, so the new test never exercises the Heze types. It works because the two are field-identical today, but the JSON structs do not set `deny_unknown_fields`, so the first Heze-only payload field will be silently dropped rather than error. Fine as a stopgap — could you leave a `TODO(heze)` on that arm noting it deliberately reuses the Gloas containers?

Related: the three `TODO(heze)` markers this PR deletes (the `get_payload` arm, `BlockProposalContentsHeze`, `get_payload_heze`) were the breadcrumbs for exactly that revisit. I would keep one rather than drop all three.

**`NewPayloadRequest::Heze` is still `UnsupportedForkVariant`**, but unreachable, since `payload_notifier.rs` always builds `NewPayloadRequest::Gloas` (`ExecutionPayloadEnvelope` hardcodes `ExecutionPayloadGloas`). So a Heze block envelope goes to a real EL over `engine_newPayloadV5`. Not introduced here and harmless while Heze is identical to Gloas — just worth being explicit in the description that this enables block production, not the Heze engine API.

**Test nits.** The comment says 'Build a full epoch past the Heze boundary', but the slot list ends at the first slot of `heze_fork_epoch + 1`, i.e. one Heze epoch plus one slot; and `assert_eq!(head_state.current_epoch(), heze_fork_epoch + 1)` is then tautological. Good that it covers the Gloas→Heze transition on the way through, and that it asserts the envelope was stored.

The duplicated Gloas/Heze arms in `complete_partial_beacon_block_gloas` look right to leave as they are, given #9729 makes them diverge anyway.

File: beacon_node/execution_layer/src/engine_api/http.rs [hunk 1]
```diff
@@ -1078,7 +1078,7 @@ impl HttpJsonRpc {
         let params = json!([JsonPayloadIdRequest::from(payload_id)]);
 
         match fork_name {
-            ForkName::Gloas => {
+            ForkName::Gloas | ForkName::Heze => {
```
Comment (dapplion) [inline beacon_node/execution_layer/src/engine_api/http.rs:1081]: 🤖 automated (dapplion's agent): this arm deserializes the Heze response into `JsonGetPayloadResponseGloas`, so it comes back tagged `GetPayloadResponse::Gloas` and flows on as `ExecutionPayloadGloas`. That leaves the already-existing `JsonGetPayloadResponseHeze` and `GetPayloadResponseHeze` unreachable on the client side, and since these JSON structs do not set `deny_unknown_fields`, the first Heze-only payload field will be silently dropped rather than error. Fine while the two are field-identical, but worth keeping a `TODO(heze)` right here recording that the Gloas containers are being reused deliberately.

Comment (conache) [reply beacon_node/execution_layer/src/engine_api/http.rs:1081]: Addressed in https://github.com/sigp/lighthouse/pull/9714/commits/db57fb78b856ff20b1d2d1d40f6c9aa122c7b88c

File: beacon_node/beacon_chain/src/block_production/gloas.rs [hunk 2]
```diff
@@ -649,13 +649,31 @@ impl<T: BeaconChainTypes> BeaconChain<T> {
...
                 },
             }),
-            // TODO(heze): construct a `BeaconBlockHeze` here once Heze block production is
-            // wired up end-to-end (get_payload, envelope handling, etc).
-            BeaconState::Heze(_) => {
-                return Err(BlockProductionError::InvalidBlockVariant(
-                    "Block production disabled for Heze".to_owned(),
-                ));
-            }
+            BeaconState::Heze(_) => BeaconBlock::Heze(BeaconBlockHeze {
```
Comment (chong-he) [inline beacon_node/beacon_chain/src/block_production/gloas.rs:755]: The function name (and also file name) for this is: `complete_partial_beacon_block_gloas` and it seems to specifically catered to Gloas at the moment, so I am not sure about adding a Heze here without modifying the function name. 

Not sure what is the best way to do about this though: 
- keeping as it is?
- new function / file name for Heze? 

Maybe @eserilev will have a better insight about this

Comment (conache) [reply beacon_node/beacon_chain/src/block_production/gloas.rs:755]: Thank you @chong-he for the feedback on this! I've also been thinking about this while working on the implementation - here are some thoughts/options I took into account:

- first option was to rename the function to `_post_gloas` instead of `_gloas` (maybe adding more clarity). However, for consistency, that would mean renaming the entire parent call chain (`produce_block_with_verification_gloas`, `produce_block_on_state_gloas`, etc.). This might imply a bigger change radius, so - to keep the PR contained - I didn't pursue this option. It could be reasonable to addopt this solution though...
- another option was to isolate [the beacon block building logic](https://github.com/conache/lighthouse/blob/heze-block-production/beacon_node/beacon_chain/src/block_production/gloas.rs#L615-L677) into two variants, a `_heze` and a `_gloas` helper. However, this doesn't solve the problem, because the parent caller - prefixed with `_gloas` - would have to stay shared anyway
- I didn't consider a newly dedicated Heze file though, my concern here would be with duplication, which would make the code harder to maintain. I'm not really sure if I miss something here

I ended up choosing to integrate the changes in the `_gloas` as is, my thinking being that the `_gloas` suffix actually means post-Gloas behaviour rather than fork-speciffic behavior (hence the renaming suggestion above seems reasonable to me)

Not sure what's best either, but happy to do the rename here or in a dedicated follow-up PR if you both think it's worth it

Comment (chong-he) [reply beacon_node/beacon_chain/src/block_production/gloas.rs:755]: We will wait for the feedback from Eitan. For Gloas, there is a new file and new function dedicated to it, because the block production is so different from previous Fork. 

I am not sure how different will Heze be, maybe it is quite similar to Gloas and hence doesn't warrant a new file like Gloas does

Comment (eserilev) [reply beacon_node/beacon_chain/src/block_production/gloas.rs:755]: the reason we did that is because we want to deprecate pre-gloas block production once glamsterdam forks. we dont want a separate heze block production path

File: beacon_node/execution_layer/src/test_utils/execution_block_generator.rs [hunk 3]
```diff
@@ -898,6 +898,13 @@ impl<E: EthSpec> ExecutionBlockGenerator<E> {
                                 .push(ProgressiveVariableList::<u8>::new(tx.into()));
                         }
                     }
+                    ExecutionPayload::Heze(payload) => {
+                        for tx in Vec::from(transactions) {
+                            payload
+                                .transactions
+                                .push(ProgressiveVariableList::<u8>::new(tx.into()));
+                        }
```
Comment (eserilev) [inline beacon_node/execution_layer/src/test_utils/execution_block_generator.rs]: these are identical right? we could do `ExecutionPayload::Gloas(_) | ExecutionPayload::Heze(_)` as the match arm

**Review by eserilev — APPROVED** [review]
LGTM

**Comment by eserilev** [comment]
@mergify requeue
